tests/testthat/test-shinytest2-teal.R

```diff
@@ -0,0 +1,317 @@
+  app_with_reporter$stop()
+})
+
+test_that("show/hide hamburger works as expected", {
+  app <- TealAppDriver$new(
+    data = simple_teal_data(),
+    modules = example_module()
+  )
+
+  get_class_attributes <- function(app, selector) {
+    element <- app$get_html(selector = selector) %>%
+      rvest::read_html() %>%
+      html_nodes(selector)
+    list(
+      class = element %>%
+        rvest::html_attr("class"),
+      style = element %>%
+        rvest::html_attr("style")
+    )
+  }
+
+  primary_attrs <- get_class_attributes(app, ".teal_primary_col")
+  secondary_attrs <- get_class_attributes(app, ".teal_secondary_col")
+
+  expect_true(grepl("col-sm-9", primary_attrs$class))
+  expect_false(isTruthy(secondary_attrs$style))
+
+  app$click(selector = ".btn.action-button.filter_hamburger")
+  app$wait_for_idle(timeout = default_idle_timeout)
+  primary_attrs <- get_class_attributes(app, ".teal_primary_col")
+  secondary_attrs <- get_class_attributes(app, ".teal_secondary_col")
+
+  expect_true(grepl("col-sm-12", primary_attrs$class))
+  expect_true(grepl("display: none;", secondary_attrs$style))
+  app$stop()
+})
+
+test_that("filter panel only shows the data supplied using datanames", {
+  app <- TealAppDriver$new(
+    data = simple_teal_data(),
+    modules = modules(
+      example_module(label = "mtcars", datanames = "mtcars")
+    )
+  )
+  app$wait_for_idle(timeout = default_idle_timeout)
+
+  expect_identical(
+    app$get_active_filter_vars(),
+    "mtcars"
+  )
+  app$stop()
+})
+
+test_that("filter panel shows all the datasets when datanames is all", {
+  app <- TealAppDriver$new(
+    data = simple_teal_data(),
+    modules = modules(
+      example_module(label = "all", datanames = "all")
+    )
+  )
+  app$wait_for_idle(timeout = default_idle_timeout)
+
+  expect_identical(
+    app$get_active_filter_vars(),
+    c("iris", "mtcars")
+  )
+  app$stop()
+})
+
+test_that("filter panel is not created when datanames is NULL", {
+  app <- TealAppDriver$new(
+    data = simple_teal_data(),
+    modules = modules(
+      example_module(label = "NULL", datanames = NULL)
+    )
+  )
+  app$wait_for_idle(timeout = default_idle_timeout)
+
+  expect_identical(
+    app$get_html(".teal_secondary_col") %>%
+      rvest::read_html() %>%
+      rvest::html_node("div") %>%
+      rvest::html_attr("style"),
+    "display: none;"
+  )
+
+  app$stop()
+})
+
+test_that("all the nested teal modules are initiated as expected", {
+  app <- TealAppDriver$new(
+    data = simple_teal_data(),
+    modules = modules(
+      example_module(label = "Example Module"),
+      modules(
+        label = "Nested Modules",
+        example_module(label = "Nested 1"),
+        example_module(label = "Nested 2"),
+        modules(
+          label = "Sub Nested Modules",
+          example_module(label = "Nested 1"),
+          example_module(label = "Nested 1")
+        )
+      )
+    )
+  )
+  app_modules <- get_app_module_tabs(app)
```

Comment:
Can we move it to the object?
```suggestion
  app_modules <- app$get_app_module_tabs()
```

Reply:
`get_app_module_tabs` is a function from the test helper file. There is no real use for people to get this value once we test this inside teal. So, I did not move this inside the R6 class.
I think we can even remove this function completely and just implement this logic inside one test. WDYT?

Reply:
Ok, makes sense to keep it outside of the class 👍